commons/zenoh-core/src/lib.rs

```diff
@@ -22,9 +22,6 @@ pub mod macros;
 pub use macros::*;
 use std::future::{Future, Ready};
 pub use zenoh_macros::*;
-pub mod zresult;
```

Comment:
A reexport of zenoh-result might be helpful here, to avoid a breaking change. It could also have saved this PR a lot of lines that change the imports. @Mallets ?

Reply:
Muy only "complain" is that this would require an explicit dependency from `zenoh_core` to `zenoh_result`. Anyway, I'm adding it so everybody is happy :)